common/jenkins-slaves/README.md

```diff
@@ -2,28 +2,34 @@
 
 Hosts all the jenkins slaves that are part of the OpenDevStack distribution.
 
-All these slaves inherit from the [slave-base](https://github.com/opendevstack/ods-core/tree/master/jenkins/slave-base), and are built in the global `CD` project. 
+All these slaves inherit from the [slave-base](https://github.com/opendevstack/ods-core/tree/master/jenkins/slave-base), and are built in the global `CD` project.
 
 Inside your `jenkinsfile` you can configure which slave is used by changing the `image` property to your imagestream (e.g. in case it's in a `project`-cd namespace and not in the global `CD` one).
-```
+
+```groovy
 odsPipeline(
   image: "${dockerRegistry}/cd/jenkins-slave-maven:latest"
+)
 ```
-the ODS [jenkins shared library](https://github.com/opendevstack/ods-jenkins-shared-library) takes care about starting the `jenkins slave` during the build as pod in your `project`s `cd` namespace with the `jenkins` service account.
 
-### Currently available slaves
-1. [Maven / Gradle](maven/README.md) 
-1. [nodeJS 10](nodejs10-angular/README.md)
-1. [Scala & SBT](scala/README.md)
-1. [Python](python/README.md)
-1. [GoLang](golang/README.md)
-1. [Airflow](airflow/README.md)
+The ODS [jenkins shared library](https://github.com/opendevstack/ods-jenkins-shared-library) takes care about starting the `jenkins slave` during the build as pod in your `project`s `cd` namespace with the `jenkins` service account.
+
+## Currently Available Slaves
 
-## OCP config / installation
+1. [Airflow](airflow)
+2. [GoLang](golang)
+3. [Maven / Gradle](maven)
+4. [Node.js 10 (Angular)](nodejs10-angular)
+5. [Node.js 12](nodejs12)
```

Comment:
The nodejs12 here seems to be shipping plain nodejs12 + yarn. In comparison the nodejs10-angular is also shipping:
* Angular CLI (for provisioning angular QS via ``ng new`` and calling the usual ``ng``commands during builds within an Angular project)
* cypress including needed drivers for running in headless mode
* Chrome for running Angular's unit tests in headless mode

nodejs10-angular is pretty much tailored to be used with Angular QS but it would also perfectly work with other frontend-related projects. The e2e-cypress and fe-vue QS are also using cypress and therfore benefit from using nodejs10-angular.

Reply:
I have to say that I have never tested with `cypress` before, but I it confuse me to refer to an `*-angular` image because of this dependency, when I want to develop in `react` or `vue` 😕 
No matter what framework(s) you use, all dependencies should be in `package.json` and installed via `npm` / `yarn`. 

So, indeed, this is a blank Node.js v12.x image and should be a blank Node.js image by the definition of name. 
You have now the possibility to use it as your next `jenkins-slave-base`-image to install `cypress` or any other `dependencies` to speed up your build process.

_By the way: I think `cypress` supports running in a [continuous integration](https://docs.cypress.io/guides/guides/continuous-integration.html#Setting-up-CI) environment?!_

Reply:
> I have to say that I have never tested with `cypress` before, but it confused me all the time to refer to an `*-angular` image because of this dependency, when I want to develop in `react` or `vue` 😕

this naming has historical reasons and is definitely something that has to change ;)
I guess a name like ``nodejs-fe`` would be better.

> _By the way: I think `cypress` supports running in a [continuous integration](https://docs.cypress.io/guides/guides/continuous-integration.html#Setting-up-CI) environment?!_

true, but... https://docs.cypress.io/guides/guides/continuous-integration.html#Dependencies

Those dependencies were integrated here into the nodejs*-angular slave:
https://github.com/opendevstack/ods-quickstarters/blob/3a2c5f78be1e9b4d7dcb7ee27643f661df97e9de/common/jenkins-slaves/nodejs10-angular/Dockerfile.rhel7#L35

> You have now the possibility to use it as your next `jenkins-slave-base`-image to install `cypress` or any other `dependencies` to speed up your build process.
> 

In general this is a good idea, but especially cypress and other browser-based testing frameworks need additional dependencies on an OS level like mentioned above. Since adding those dependencies will need root rights, there is no (secure) way this could be done during a Jenkins build.

BTW: I haven't tried it yet, but it should be possible to leave Angular CLI out of the nodejs*-angular slave. Instead of running ``ng something`` from a global Angular CLI installation you can also call ``npx ng something``, which will dirctly use the version from inside your ``node_modules`` folder.
Same thing with cypress: I successfully installed it via ``package.json``without using the version, that was baked into the slave. This was not trivial, since cypress loads Chrome via post-installation during ``npm install``... proxy issues, additional hash checks of the downloaded binaries, etc. 
... and at the end of the day you would still need the OS level dependencies mentioned above ¯\\_(ツ)_/¯

Reply:
@michaelsauter & @cschweikert cypress ci dependencies are now part of the jenkins-slave-nodejs12 image. See commit: https://github.com/opendevstack/ods-quickstarters/pull/232/commits/51472994dbf62eca71d428cea5d728482c31ebd0

Reply:
Mh, mybe one important point got lost from the whole discussion around testing frameworks: From my point of view having both slaves, ``nodejs*`` and ``nodejs*-angular``, makes sense. My only remark was and still is to clearly document, what the actual difference between the two slaves is, e.g. with a fair "warning" in the ``nodejs*`` readme. Beside that everything is super fine and I'm happy to have a plain ``nodejs*`` slave without cypress, angular and chrome.

Reply:
@cschweikert I removed the cypress ci dependencies again and mention it on the `Known Limitations`